internal/pkg/enccid/enc_cid.go

```diff
@@ -0,0 +1,114 @@
+package enccid
+
+import (
+	"encoding/json"
+	"fmt"
+
+	cbor "github.com/fxamacker/cbor"
+	cid "github.com/ipfs/go-cid"
+	ipldcbor "github.com/ipfs/go-ipld-cbor"
+)
+
+// EncodableCid is a cid wrapper that implements UnmarshalCBOR and MarshalCBOR.
+type EncodableCid struct {
```

Comment:
I would rename this package to **cid** and this type to `Cid`. 
Can we also leave a TODO to submit a PR to `go-cid` to add the un/marshaling methods?

Reply:
We should sync offline about the systemic problems blocking the suggested approach